src/commoncode/fileutils.py

```diff
@@ -185,6 +185,17 @@ def get_temp_dir(base_dir=scancode_temp_dir, prefix=''):
 
 # TODO: move these functions to paths.py
 
+def prepare_path(pth):
+    """
+    Return the `pth` path string either as encoded bytes if on Linux and using
+    Python 2 or as a unicode/text otherwise.
+    """
+    if on_linux and py2 and isinstance(pth, compat.unicode):
+        return fsencode(pth)
+    if not isinstance(pth, compat.unicode):
+        return fsdecode(pth)
+    return pth
+
```

Comment:
@Abhishek-Dev09 This is wrong IMHO
We should instead have something like this:
```
    if on_linux and py2:
        if not isinstance(pth, compat.bytes):
            pth = fsencode(pth)
        return path
    if not isinstance(pth, compat.unicode):
        return fsdecode(pth)
    return pth

```

Reply:
This changes have been made by you only.

Reply:
Sure, that does not mean that you should not fix it.

Reply:
i have fixed

Reply:
You can never hurt me!